### PR TITLE
[WIP] Migrates object service dashboard tests to cypress

### PR DESCRIPTION
**Pull request by vbnrh**
Signed-off-by: Vineet Badrinath <vbadrina@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  before(() => {
+    installStorageCluster();
+  });
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.byLegacyTestID('horizontal-link-Object Service').click();
+  });
+  it('Test at least one noobaa bucket is present', () => {
+    cy.get('div:nth-child(1) > div.co-inventory-card__item-title')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:36]: Use `data-test`

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      .invoke('text')
+      .then((text) => {
+        const count = parseInt(text.split(' ')[0], 10);
+        expect(count).toBeGreaterThanOrEqual(1);
+      });
+  });
+
+  it('Create an Object Bucket Claim and test equality', () => {
+    cy.get('div:nth-child(3) > div.co-inventory-card__item-title')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:45]: Use data-test

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      .invoke('text')
+      .then((text) => {
+        const initalCount = parseInt(text.split(' ')[0], 10);
+        cy.exec(`echo '${JSON.stringify(testBucket)}' | kubectl create -f -`);
+        cy.get('div:nth-child(3) > div.co-inventory-card__item-title')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:50]: use data-test

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    const RGW_PROVISIONER = 'openshift-storage.ceph.rook.io/bucket';
+    cy.exec('oc get StorageClasses -o json').then((result) => {
+      if (result.code === 0) {
+        const storageClasses = JSON.parse(result.stdout.toString());
+        return storageClasses.items.some((sc) => sc.provisioner === RGW_PROVISIONER);
+      }
+      return false;
+    });
+    return false;
+  };
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:72]: Move it to utils.

Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:72]: We are always running this test on cloud at this current point so we don't really need this. We can introduce this once we need it.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  });
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.byLegacyTestID('horizontal-link-Object Service').click();
+  });
+
+  it('Check if Multi Cloud Gateway is in a healthy state', () => {
+    if (!isRGWPresent()) {
+      cy.get('div > div > div:nth-child(1) > div > div.co-dashboard-icon > svg').should(
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:84]: data-test

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  });
+
+  it('Check if Data Resiliency of MCG is in healthy state', () => {
+    if (!isRGWPresent()) {
+      cy.get('div > div > div:nth-child(2) > div > div.co-dashboard-icon > svg').should(
+        'have.attr',
+        'data-test',
+        'success-icon',
+      );
+    }
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:99]:
```suggestion
```
RGW is not going to be present.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    installStorageCluster();
+  });
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.byLegacyTestID('horizontal-link-Object Service').click();
+  });
+
+  it('Check if Efficiency Ratio is in acceptable data range', () => {
+    cy.get('div:nth-child(1) > div.ceph-storage-efficiency-card__item-status > span')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:114]: data-test

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+          expect(text).toEqual('Not available');
+        } else {
+          expect(numA).toBeGreaterThan(0);
+          expect(numB).toEqual(1);
+        }
+      });
+  });
+
+  it('Check for savings value to be in acceptable data range', () => {
+    cy.get('div:nth-child(2) > div.ceph-storage-efficiency-card__item-status > span')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:129]: data-test

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    installStorageCluster();
+  });
+
+  beforeEach(() => {
+    cy.visit('/');
+    cy.byLegacyTestID('horizontal-link-Object Service').click();
+  });
+
+  it('Check if resource provider has at least 1 provider', () => {
+    cy.get('.nb-resource-providers-card__row-title')
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/object-service-dashboards.spec.ts:154]: data-test

**Review by bipuladh — CHANGES_REQUESTED** [review]
skimmed through it, needs a more indepth review.

**Comment by rexagod** [comment]
^^ Wanted to look at CI's artifacts and whether they differ from my local results.

**Comment by rexagod** [comment]
Continued @ #8270 since I don't have permissions to push here.

**Pull request closed without merging**